drivers/interrupt_controller/intc_plic.c

```diff
@@ -48,18 +50,47 @@ static inline uint32_t get_plic_enabled_size(const struct device *dev)
 	return (config->num_irqs >> 5) + 1;
 }
 
+/* Forward declaration */
+#define PLIC_INTC_DEV_CONFIG_DECL(n) static const struct plic_config plic_config_##n
+DT_INST_FOREACH_STATUS_OKAY(PLIC_INTC_DEV_CONFIG_DECL);
+
+#define PLIC_GET_DEV_FROM_IRQ(n)                                                                   \
+	do {                                                                                       \
+		if ((irq >= plic_config_##n.isr_offset) &&                                         \
+		    (irq < (plic_config_##n.isr_offset + plic_config_##n.num_irqs))) {             \
+			return DEVICE_DT_INST_GET(n);                                              \
+		}                                                                                  \
+	} while (false)
```

Comment:
is the `do ... while` really needed here?

Reply:
We have quite a lot of these in Zephyr and my impression is that it will be optimized by the compiler and it existed to ensure that the invocation of the macro to be `;` terminated

Do you prefer the `do ... while` to be excluded from the macro instead?

Reply:
leave them if they are actually needed